diff --git a/overlays/bootstrap.nix b/overlays/bootstrap.nix
index b2e66de3..c1078977 100644
--- a/overlays/bootstrap.nix
+++ b/overlays/bootstrap.nix
@@ -168,6 +168,10 @@ in {
                 ++ fromUntil "8.10.5" "8.10.6" ./patches/ghc/ghc-8.10.5-add-rts-exports.patch
                 ++ final.lib.optionals final.hostPlatform.isDarwin
                   (fromUntil "8.10.5" "8.10.6" ./patches/ghc/ghc-8.10.5-darwin-allocateExec.patch)
+                ++ always (final.fetchpatch {
+                            url = "https://raw.githubusercontent.com/NixOS/nixpkgs/a8c2421e5b3652b9ff6db1856382d7e0dc8b97df/pkgs/development/compilers/ghc/sphinx-4-configuration.patch";
+                            sha256 = "sha256-r19nZChdsUFHIABH/M0DO2z78/Eo+P/Ix89ToDABjn8=";
+                          })
                 ;
         in ({
             ghc844 = final.callPackage ../compiler/ghc {
